chalicelib/lib/helpers.py

```diff
@@ -25,17 +28,64 @@ def date_range(start_date, stop_date):
 
 
 def validate_date(date, format_str) -> bool:
+    """
+    Validate a string in date format is valid
+    """
 
     try:
-        if ":" in date:
-            start, stop = date.split(":")
-            datetime.strptime(start, format_str)
-            datetime.strptime(stop, format_str)
-            if start > stop:
-                return False
-        else:
-            datetime.strptime(date, format_str)
-    except TypeError:
+        datetime.strptime(date, format_str)
+    except (TypeError, ValueError) as error:
+        log.error(f"Unable to validate date {date}. Error was: {error}")
         return False
 
     return True
+
+
+def parse_date(date: str, format_str: str = "%Y-%m-%d") -> dict:
+    """
+    Parse the date from a string.
+    Expects these formats:
+    "today"
+    "2019-01-01"
+    "2019-01-01:2019-01-02"
+    """
+
+    dates = dict()
```

Comment:
Do we really need a dict here? Once we have datetime objects we can retrieve the strftime easily with object.strftime(format_str)

Reply:
So rather a list with only datetime and then the caller needs to use the correct format string to get the string representation?

Reply:
Pushed the change so parse_date now returns list of datetime objects